### PR TITLE
Remove cmd/gsutil image

### PR DESCRIPTION
**Pull request by ImJasonH**
Based on #1503

This image wrapped any image with `gsutil` in the path (typically,
`google/cloud-sdk`), and invoked it with our own custom Go wrapper.
That's unnecessary, since `gsutil` can be invoked directly as necessary,
using any image where `gsutil` is in the PATH (`google/cloud-sdk` is the
default).

This change should be entirely transparent to end users.

Operators can override the image being used if they so desire, to point
to any image that contains `gsutil`.

Historical note: `cmd/gsutil` was added in https://github.com/tektoncd/pipeline/pull/321, ostensibly to fix https://github.com/tektoncd/pipeline/issues/124, though the specific rationale for the wrapper script might be lost to time. 🕥 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Use google/cloud-sdk directly to invoke gsutil in support of internal operations, instead of our own wrapper around that image.
```

### CODE REVIEW

File: pkg/apis/pipeline/v1alpha1/artifact_bucket.go-e [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (vdemeester) [inline pkg/apis/pipeline/v1alpha1/artifact_bucket.go-e]: `go-e` ? :thinking:

Comment (ImJasonH) [reply pkg/apis/pipeline/v1alpha1/artifact_bucket.go-e]: Whooooops! Gone now. 🙈

**Comment by ImJasonH** [comment]
/test pull-tekton-pipeline-integration-tests

❄️

**Comment by sbwsg** [comment]
Looks like our gsutil wrapper may have been introduced to solve an issue similar to [this one](https://github.com/GoogleCloudPlatform/cloud-sdk-docker/issues/152).

We had an explicit issue around this [here](https://github.com/tektoncd/pipeline/issues/402)

**Comment by ImJasonH** [comment]
It sounds like gcloud expects users to call `gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS` before making calls.

We shouldn't introduce an image to run this for users, I think. Instead we should document this and make sure users know that gcloud expects this of them.

**Comment by sbwsg** [comment]
Sounds good to me.  We... probably need to consider this a breaking change in 0.9 tho right?  I'll update the release notes.

**Comment by ImJasonH** [comment]
Yeah. If we guarantee this behavior going forward we should add a test.

**Pull request merged**